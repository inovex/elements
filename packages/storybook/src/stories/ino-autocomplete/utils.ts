import { KeyValue } from '@inovex.de/elements';

export const AutoCompleteStoryOptions: KeyValue[] = [
  {
    key: 'hh',
    value: 'Hamburg',
  },
  {
    key: 'ber',
    value: 'Berlin',
  },
  {
    key: 'k',
    value: 'Karlsruhe',
  },
];
