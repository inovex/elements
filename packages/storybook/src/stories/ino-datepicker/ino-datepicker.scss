.story-datepicker {
  .customizable-picker {
    margin-bottom: 48px;
    width: 340px;
  }

  h4 {
    margin-top: 32px;
  }
}

.datepicker-group {
  display: flex;
  width: 630px;
  height: 550px;
  background: white;
  border-radius: 10px;
  box-shadow: 0 25px 50px -12px rgba(#1e2125, 0.25);

  aside {
    width: 230px;
    background: #eff5fc;
    height: 100%;
    border-top-left-radius: 10px;
    border-bottom-left-radius: 10px;
  }
  main {
    padding: 24px;
  }

  li {
    list-style: none;
    padding: 12px 24px;
    cursor: pointer;
    font-weight: 500;
    font-size: 14px;
  }
  ul {
    padding: 0;
  }

  ino-radio-group {
    flex-grow: 1;
    display: flex;
    justify-content: space-between;
    margin: 0 12px 24px 0;
  }

  ino-input {
    margin-bottom: 24px;
  }
}

.fancy-datepicker {
  width: 350px;
}
