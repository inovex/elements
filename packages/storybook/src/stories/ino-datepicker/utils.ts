import '@inovex.de/elements';

export function registerInlineDatepickerHandler() {
  const group = document.querySelector('ino-radio-group') as HTMLInoRadioGroupElement;
  const datepicker = document.querySelector('.datepicker-group ino-datepicker') as HTMLInoDatepickerElement;

  const today = document.querySelector('.today');
  const thisWeek = document.querySelector('.thisWeek');
  const lastWeek = document.querySelector('.lastWeek');

  const todayClickHandler = () => {
    datepicker.value = '02.03.2020';
    datepicker.label = 'Am';
    group.value = 'at';
    datepicker.range = false;
  };

  const thisWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '02.03.2020 bis 08.03.2020';
  };

  const lastWeekClickHandler = () => {
    group.value = 'range';
    datepicker.label = 'Zeitraum';
    datepicker.range = true;
    datepicker.value = '24.02.2020 bis 01.03.2020';
  };

  const checkedChangeHandler = (e: CustomEvent<boolean>) => {
    const target = e.target as HTMLInoRadioElement;

    switch (target.value) {
      case 'at':
        datepicker.label = 'Am';
        datepicker.range = false;
        break;
      case 'after':
        datepicker.label = 'Ab';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').shift();
        }
        break;
      case 'before':
        datepicker.label = 'Bis';
        datepicker.range = false;
        if (group.value == 'range') {
          datepicker.value = datepicker.value.split(' - ').pop();
        }
        break;
      case 'range':
        datepicker.range = true;
        datepicker.label = 'Zeitraum';
        break;
    }
    group.value = target.value;
  };

  today.addEventListener('click', todayClickHandler);
  thisWeek.addEventListener('click', thisWeekClickHandler);
  lastWeek.addEventListener('click', lastWeekClickHandler);
  group.addEventListener('checkedChange', checkedChangeHandler);

  return () => {
    today.removeEventListener('click', todayClickHandler);
    thisWeek.removeEventListener('click', thisWeekClickHandler);
    lastWeek.removeEventListener('click', lastWeekClickHandler);
    group.removeEventListener('checkedChange', checkedChangeHandler);
  };
}
