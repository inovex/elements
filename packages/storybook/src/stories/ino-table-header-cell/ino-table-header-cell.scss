.story-ino-table-header-cell {
  ino-table {
    display: block;
    margin-bottom: 40px;
  }

  .ino-table-header-cell__popover-content ino-input {
    padding: 0 20px 20px;
  }

  ino-table-header-cell {
    min-width: 230px;
  }
}
