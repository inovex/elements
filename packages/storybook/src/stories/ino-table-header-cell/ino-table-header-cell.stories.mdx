import { Meta, Description } from "@storybook/addon-docs";
import readme from "../../../../elements/src/components/ino-table-header-cell/readme.md";

{/* <Meta title="Structure/ino-table-header-cell/Documentation" /> */}

<Description markdown={readme} />
