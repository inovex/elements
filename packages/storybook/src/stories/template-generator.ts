import { Story } from '@storybook/web-components';
import docsJson from '../../elements-stencil-docs';

export class TemplateGenerator<Component> {
  private readonly tagName: string;
  private readonly templateFn: Story<Component>;

  constructor(tagName: string, templateFn: Story<Component>) {
    this.tagName = tagName;
    this.templateFn = templateFn;
  }

  public generateStory(args: Component, prop?: string) {
    const BoundTemplate: Story<Component> = this.templateFn.bind({});
    BoundTemplate.args = { ...args };

    if (prop) {
      BoundTemplate.parameters = {
        docs: {
          description: {
            story: this.findPropertyDocs(prop),
          },
        },
      };
    }

    return BoundTemplate;
  }

  private findPropertyDocs(prop: string): string {
    return docsJson.components
      .find((component) => component.tag === this.tagName)
      .props.find((p) => p.name === prop).docs;
  }
}
