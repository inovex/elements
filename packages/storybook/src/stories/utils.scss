/**
 * Applies styling to the container of a story.
 * The styling is encapsulated within this mixin because the ID may change in the future.
 */
@mixin story-container {
  #root-inner {
    @content;
  }
}
