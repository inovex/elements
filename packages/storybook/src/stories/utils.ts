import { html } from 'lit-html';
import ICONS from '../../../elements/src/components/ino-icon/icons';

export const maybeCreateStoryArgs = (story: any) => {
  if (!story.args) {
    story.args = {};
  }
  if (!story.argTypes) {
    story.argTypes = {};
  }
};

export const withIconControl = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: getIcons(),
  };
};

export const withColorScheme = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    control: {
      type: 'select',
    },
    options: [
      'primary',
      'secondary',
      'success',
      'warning',
      'error',
      'light',
      'dark',
    ],
  };
};

export const withSortDirection = (
  story: any,
  propertyName: string,
  defaultValue?: string
) => {
  maybeCreateStoryArgs(story);
  story.args[propertyName] = defaultValue || '';
  story.argTypes[propertyName] = {
    options: ['desc', 'asc'],
    control: {
      type: 'select',
    },
  };
};

export const getIcons = () => ICONS.filter((icon) => !icon.startsWith('_'));

export const defaultDecorator = (story, className?: string) => {
  return html` <div class="${className ?? ''}">${story()}</div> `;
};
