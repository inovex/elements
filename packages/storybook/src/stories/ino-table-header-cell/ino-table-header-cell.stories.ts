import { html } from 'lit-html';
import { decorateStoryWithClass, withIconControl } from '../utils';
import { Meta } from '@storybook/web-components';
import { useEffect } from '@storybook/client-api';
import { Components } from '@inovex.de/elements';
import { TemplateGenerator } from '../template-generator';
import './ino-table-header-cell.scss';
import { registerInlineDatepickerHandler } from '../ino-datepicker/utils';

export default {
  title: `Structure/ino-table-header-cell`,
  component: 'ino-table-header-cell',
  decorators: [
    (story) => decorateStoryWithClass(story, 'story-ino-table-header-cell'),
    (story) => {
      useEffect(() => {
        const sortChangeHandler = (e) => {
          const inoTable = e.target as Components.InoTable;
          const { columnId, sortDirection } = e.detail;
          inoTable.sortColumnId = columnId;
          inoTable.sortDirection = sortDirection;
        };
        document.addEventListener('sortChange', sortChangeHandler);

        const valueChangedHandler = (e) => (e.target.value = e.detail);
        const inoInputs = document.querySelectorAll('ino-input');
        inoInputs.forEach((el) =>
          el.addEventListener('valueChange', valueChangedHandler),
        );

        return () => {
          document.removeEventListener('sortChange', sortChangeHandler);
          inoInputs.forEach((el) =>
            el.removeEventListener('valueChange', valueChangedHandler),
          );
        };
      });
      return story();
    },
  ],
  args: {
    autofocus: false,
    notSortable: false,
    label: 'Label',
    columnId: '',
    sortDirection: '',
    searched: false,
    searchIcon: 'search',
  },
} as Meta;

const template = new TemplateGenerator<Components.InoTableHeaderCell>(
  'ino-table-header-cell',
  (args) => html`
    <ino-table>
      <ino-table-header-cell
        autofocus="${args.autofocus}"
        not-sortable="${args.notSortable}"
        sort-direction="${args.sortDirection}"
        search-icon="${args.searchIcon}"
        searched="${args.searched}"
        label="${args.label}"
      >
        <ino-input placeholder="Search for XY..." data-ino-focus>
          <ino-icon clickable slot="icon-trailing" icon="close"></ino-icon>
        </ino-input>
      </ino-table-header-cell>
      </tr>
    </ino-table>
  `,
);

export const Playground = template.generatePlaygroundStory();

withIconControl(Playground, 'searchIcon', 'search');

export const Label = template.generateStoryForProp('label', 'Search for XY...');
export const SortDirection = template.generateStoryForProp(
  'sortDirection',
  'desc',
);
export const NotSortable = template.generateStoryForProp('notSortable', true);
export const SearchIcon = template.generateStoryForProp('searchIcon', 'search');
export const Searched = template.generateStoryForProp('searched', true);

const templateSearchWithDatepicker =
  new TemplateGenerator<Components.InoTableHeaderCell>(
    'ino-table-header-cell',
    () => html`
      ${useEffect(registerInlineDatepickerHandler)}
      <ino-table>
        <tr slot="header-row">
          <ino-table-header-cell
            label="Column Selection Search"
            search-icon="calendar"
            class="datepicker-cell"
          >
            <div lang="de" class="datepicker-group">
              <aside>
                <ul>
                  <li class="today">Today</li>
                  <li class="lastWeek">last week</li>
                  <li class="thisWeek">this week</li>
                </ul>
              </aside>
              <main id="main">
                <ino-datepicker
                  type="date"
                  label="An"
                  inline
                  date-format="d.m.Y"
                  placeholder="tt.mm.jjjj"
                  append-to="main"
                  data-ino-focus
                ></ino-datepicker>
                <ino-radio-group value="at">
                  <ino-radio value="at">At</ino-radio>
                  <ino-radio value="after">After</ino-radio>
                  <ino-radio value="before">Before</ino-radio>
                  <ino-radio value="range">Range</ino-radio>
                </ino-radio-group>
              </main>
            </div>
          </ino-table-header-cell>
        </tr>
      </ino-table>
    `,
  );

/**
 * Use a `ino-datepicker` element inside a `ino-table-header-cell` element to show a search with a datepicker on click
 */
export const SearchWithDatepicker =
  templateSearchWithDatepicker.generatePlaygroundStory();

const templateSearchWithSelection =
  new TemplateGenerator<Components.InoTableHeaderCell>(
    'ino-table-header-cell',
    () => {
      useEffect(() => {
        const checkboxes = document.querySelectorAll('ino-checkbox');
        const checkedHandler = (e) =>
          ((e.target as Components.InoCheckbox).checked = (
            e as CustomEvent
          ).detail);
        checkboxes.forEach(() =>
          addEventListener('checkedChange', checkedHandler),
        );
        return () =>
          checkboxes.forEach(() =>
            removeEventListener('checkedChange', checkedHandler),
          );
      });

      return html`
        <ino-table>
          <tr slot="header-row">
            <ino-table-header-cell
              label="Column Selection Search"
              search-icon="filter"
            >
              <ino-list>
                <ino-list-item text="Option 1">
                  <ino-checkbox slot="leading" selection></ino-checkbox>
                </ino-list-item>
                <ino-list-item-divider inset></ino-list-item-divider>
                <ino-list-item text="Option 2">
                  <ino-checkbox slot="leading" selection></ino-checkbox>
                </ino-list-item>
                <ino-list-item-divider inset></ino-list-item-divider>
                <ino-list-item text="Option 3">
                  <ino-checkbox slot="leading" selection></ino-checkbox>
                </ino-list-item>
                <ino-list-item-divider inset></ino-list-item-divider>
                <ino-list-item text="Option 4">
                  <ino-checkbox slot="leading" selection></ino-checkbox>
                </ino-list-item>
                <ino-list-item-divider inset></ino-list-item-divider>
              </ino-list>
            </ino-table-header-cell>
          </tr>
        </ino-table>
      `;
    },
  );

/**
 * Use a `ino-list` element inside of a `ino-table-header-cell` element to show a list selection on click
 */
export const SearchWithSelection =
  templateSearchWithSelection.generatePlaygroundStory();
