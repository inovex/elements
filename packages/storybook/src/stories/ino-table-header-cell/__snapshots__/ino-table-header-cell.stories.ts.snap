// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Structure/ino-table-header-cell Label smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-table-header-cell">
    <ino-table class="hydrated">
      <div class="mdc-data-table ino-table--no-header">
        <div class="mdc-data-table__table-container">
          <table class="ino-table__table mdc-data-table__table">
            <tbody class="mdc-data-table__content">
              <ino-table-header-cell autofocus="false"
                                     not-sortable="false"
                                     sort-direction
                                     search-icon="search"
                                     cell-id="some-cell-id"
                                     column-id="some-random-id"
                                     searched="false"
                                     label="Search for XY..."
                                     role="columnheader"
                                     class="mdc-data-table__header-cell ino-table-header-cell--searchable ino-table-header-cell--sortable hydrated"
                                     data-column-id="some-random-id"
                                     tabindex="-1"
                                     id="some-cell-id"
                                     style="max-width: 230px;"
              >
                <ino-popover color-scheme="transparent"
                             class="hydrated"
                >
                  <div class="mdc-data-table__header-cell-wrapper"
                       slot="popover-trigger"
                       tabindex="0"
                       aria-expanded="false"
                  >
                    <ino-icon class="ino-table-header-cell__search-icon ino-icon--color-scheme-primary hydrated">
                    </ino-icon>
                    <div class="mdc-data-table__header-cell-label">
                      Search for XY...
                    </div>
                  </div>
                  <div>
                  </div>
                </ino-popover>
                <ino-icon-button class="sort-none ino-icon-button--color-scheme-dark hydrated">
                  <button class="mdc-icon-button mdc-ripple-upgraded--unbounded mdc-ripple-upgraded"
                          type="button"
                          style="--mdc-ripple-fg-size:14px; --mdc-ripple-fg-scale:1.71429; --mdc-ripple-left:5px; --mdc-ripple-top:5px;"
                  >
                    <div class="mdc-icon-button__ripple">
                    </div>
                    <span class="mdc-icon-button__icon">
                      <ino-icon class="ino-icon--color-scheme-primary hydrated">
                      </ino-icon>
                    </span>
                  </button>
                </ino-icon-button>
              </ino-table-header-cell>
            </tbody>
          </table>
        </div>
        <div class="mdc-data-table__progress-indicator">
          <div class="mdc-data-table__scrim">
          </div>
        </div>
      </div>
    </ino-table>
  </div>
</div>
`;

exports[`Structure/ino-table-header-cell NotSortable smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-table-header-cell">
    <ino-table class="hydrated">
      <div class="mdc-data-table ino-table--no-header">
        <div class="mdc-data-table__table-container">
          <table class="ino-table__table mdc-data-table__table">
            <tbody class="mdc-data-table__content">
              <ino-table-header-cell autofocus="false"
                                     not-sortable="true"
                                     sort-direction
                                     search-icon="search"
                                     cell-id="some-cell-id"
                                     column-id="some-random-id"
                                     searched="false"
                                     label="Label"
                                     role="columnheader"
                                     class="mdc-data-table__header-cell ino-table-header-cell--searchable hydrated"
                                     data-column-id="some-random-id"
                                     tabindex="-1"
                                     id="some-cell-id"
                                     style="max-width: 230px;"
              >
                <ino-popover color-scheme="transparent"
                             class="hydrated"
                >
                  <div class="mdc-data-table__header-cell-wrapper"
                       slot="popover-trigger"
                       tabindex="0"
                       aria-expanded="false"
                  >
                    <ino-icon class="ino-table-header-cell__search-icon ino-icon--color-scheme-primary hydrated">
                    </ino-icon>
                    <div class="mdc-data-table__header-cell-label">
                      Label
                    </div>
                  </div>
                  <div>
                  </div>
                </ino-popover>
              </ino-table-header-cell>
            </tbody>
          </table>
        </div>
        <div class="mdc-data-table__progress-indicator">
          <div class="mdc-data-table__scrim">
          </div>
        </div>
      </div>
    </ino-table>
  </div>
</div>
`;

exports[`Structure/ino-table-header-cell Playground smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-table-header-cell">
    <ino-table class="hydrated">
      <div class="mdc-data-table ino-table--no-header">
        <div class="mdc-data-table__table-container">
          <table class="ino-table__table mdc-data-table__table">
            <tbody class="mdc-data-table__content">
              <ino-table-header-cell autofocus="false"
                                     not-sortable="false"
                                     sort-direction
                                     search-icon="search"
                                     cell-id="some-cell-id"
                                     column-id="some-random-id"
                                     searched="false"
                                     label="Label"
                                     role="columnheader"
                                     class="mdc-data-table__header-cell ino-table-header-cell--searchable ino-table-header-cell--sortable hydrated"
                                     data-column-id="some-random-id"
                                     tabindex="-1"
                                     id="some-cell-id"
                                     style="max-width: 230px;"
              >
                <ino-popover color-scheme="transparent"
                             class="hydrated"
                >
                  <div class="mdc-data-table__header-cell-wrapper"
                       slot="popover-trigger"
                       tabindex="0"
                       aria-expanded="false"
                  >
                    <ino-icon class="ino-table-header-cell__search-icon hydrated ino-icon--color-scheme-primary">
                    </ino-icon>
                    <div class="mdc-data-table__header-cell-label">
                      Label
                    </div>
                  </div>
                  <div>
                  </div>
                </ino-popover>
                <ino-icon-button class="sort-none ino-icon-button--color-scheme-dark hydrated">
                  <button class="mdc-icon-button mdc-ripple-upgraded--unbounded mdc-ripple-upgraded"
                          type="button"
                          style="--mdc-ripple-fg-size:14px; --mdc-ripple-fg-scale:1.71429; --mdc-ripple-left:5px; --mdc-ripple-top:5px;"
                  >
                    <div class="mdc-icon-button__ripple">
                    </div>
                    <span class="mdc-icon-button__icon">
                      <ino-icon class="hydrated ino-icon--color-scheme-primary">
                      </ino-icon>
                    </span>
                  </button>
                </ino-icon-button>
              </ino-table-header-cell>
            </tbody>
          </table>
        </div>
        <div class="mdc-data-table__progress-indicator">
          <div class="mdc-data-table__scrim">
          </div>
        </div>
      </div>
    </ino-table>
  </div>
</div>
`;

exports[`Structure/ino-table-header-cell SearchIcon smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-table-header-cell">
    <ino-table class="hydrated">
      <div class="mdc-data-table ino-table--no-header">
        <div class="mdc-data-table__table-container">
          <table class="ino-table__table mdc-data-table__table">
            <tbody class="mdc-data-table__content">
              <ino-table-header-cell autofocus="false"
                                     not-sortable="false"
                                     sort-direction
                                     search-icon="search"
                                     cell-id="some-cell-id"
                                     column-id="some-random-id"
                                     searched="false"
                                     label="Label"
                                     role="columnheader"
                                     class="mdc-data-table__header-cell ino-table-header-cell--searchable ino-table-header-cell--sortable hydrated"
                                     data-column-id="some-random-id"
                                     tabindex="-1"
                                     id="some-cell-id"
                                     style="max-width: 230px;"
              >
                <ino-popover color-scheme="transparent"
                             class="hydrated"
                >
                  <div class="mdc-data-table__header-cell-wrapper"
                       slot="popover-trigger"
                       tabindex="0"
                       aria-expanded="false"
                  >
                    <ino-icon class="ino-table-header-cell__search-icon ino-icon--color-scheme-primary hydrated">
                    </ino-icon>
                    <div class="mdc-data-table__header-cell-label">
                      Label
                    </div>
                  </div>
                  <div>
                  </div>
                </ino-popover>
                <ino-icon-button class="sort-none ino-icon-button--color-scheme-dark hydrated">
                  <button class="mdc-icon-button mdc-ripple-upgraded--unbounded mdc-ripple-upgraded"
                          type="button"
                          style="--mdc-ripple-fg-size:14px; --mdc-ripple-fg-scale:1.71429; --mdc-ripple-left:5px; --mdc-ripple-top:5px;"
                  >
                    <div class="mdc-icon-button__ripple">
                    </div>
                    <span class="mdc-icon-button__icon">
                      <ino-icon class="ino-icon--color-scheme-primary hydrated">
                      </ino-icon>
                    </span>
                  </button>
                </ino-icon-button>
              </ino-table-header-cell>
            </tbody>
          </table>
        </div>
        <div class="mdc-data-table__progress-indicator">
          <div class="mdc-data-table__scrim">
          </div>
        </div>
      </div>
    </ino-table>
  </div>
</div>
`;

exports[`Structure/ino-table-header-cell SearchWithDatepicker smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-table-header-cell">
    <ino-table class="hydrated">
      <div class="mdc-data-table ino-table--no-header">
        <div class="mdc-data-table__table-container">
          <table class="ino-table__table mdc-data-table__table">
            <tbody class="mdc-data-table__content">
              <ino-table-header-cell label="Column Selection Search"
                                     search-icon="calendar"
                                     class="datepicker-cell mdc-data-table__header-cell ino-table-header-cell--searchable ino-table-header-cell--sortable hydrated"
                                     cell-id="datepicker-cell"
                                     column-id="datepicker-cell-col-id"
                                     role="columnheader"
                                     data-column-id="datepicker-cell-col-id"
                                     tabindex="-1"
                                     id="datepicker-cell"
                                     style="max-width: 230px;"
              >
                <ino-popover color-scheme="transparent"
                             class="hydrated"
                >
                  <div class="mdc-data-table__header-cell-wrapper"
                       slot="popover-trigger"
                       tabindex="0"
                       aria-expanded="false"
                  >
                    <ino-icon class="ino-table-header-cell__search-icon hydrated ino-icon--color-scheme-primary">
                    </ino-icon>
                    <div class="mdc-data-table__header-cell-label">
                      Column Selection Search
                    </div>
                  </div>
                  <div>
                  </div>
                </ino-popover>
                <ino-icon-button class="sort-none ino-icon-button--color-scheme-dark hydrated">
                  <button class="mdc-icon-button mdc-ripple-upgraded--unbounded mdc-ripple-upgraded"
                          type="button"
                          style="--mdc-ripple-fg-size:14px; --mdc-ripple-fg-scale:1.71429; --mdc-ripple-left:5px; --mdc-ripple-top:5px;"
                  >
                    <div class="mdc-icon-button__ripple">
                    </div>
                    <span class="mdc-icon-button__icon">
                      <ino-icon class="ino-icon--color-scheme-primary hydrated">
                      </ino-icon>
                    </span>
                  </button>
                </ino-icon-button>
              </ino-table-header-cell>
            </tbody>
          </table>
        </div>
        <div class="mdc-data-table__progress-indicator">
          <div class="mdc-data-table__scrim">
          </div>
        </div>
      </div>
    </ino-table>
  </div>
</div>
`;

exports[`Structure/ino-table-header-cell SearchWithSelection smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-table-header-cell">
    <ino-table class="hydrated">
      <div class="mdc-data-table ino-table--no-header">
        <div class="mdc-data-table__table-container">
          <table class="ino-table__table mdc-data-table__table">
            <tbody class="mdc-data-table__content">
              <ino-table-header-cell label="Column Selection Search"
                                     search-icon="filter"
                                     cell-id="cell-id"
                                     column-id="cell-col-id"
                                     role="columnheader"
                                     class="mdc-data-table__header-cell ino-table-header-cell--searchable ino-table-header-cell--sortable hydrated"
                                     data-column-id="cell-col-id"
                                     tabindex="-1"
                                     id="cell-id"
                                     style="max-width: 230px;"
              >
                <ino-popover color-scheme="transparent"
                             class="hydrated"
                >
                  <div class="mdc-data-table__header-cell-wrapper"
                       slot="popover-trigger"
                       tabindex="0"
                       aria-expanded="false"
                  >
                    <ino-icon class="ino-table-header-cell__search-icon hydrated ino-icon--color-scheme-primary">
                    </ino-icon>
                    <div class="mdc-data-table__header-cell-label">
                      Column Selection Search
                    </div>
                  </div>
                  <div>
                  </div>
                </ino-popover>
                <ino-icon-button class="sort-none ino-icon-button--color-scheme-dark hydrated">
                  <button class="mdc-icon-button mdc-ripple-upgraded--unbounded mdc-ripple-upgraded"
                          type="button"
                          style="--mdc-ripple-fg-size:14px; --mdc-ripple-fg-scale:1.71429; --mdc-ripple-left:5px; --mdc-ripple-top:5px;"
                  >
                    <div class="mdc-icon-button__ripple">
                    </div>
                    <span class="mdc-icon-button__icon">
                      <ino-icon class="ino-icon--color-scheme-primary hydrated">
                      </ino-icon>
                    </span>
                  </button>
                </ino-icon-button>
              </ino-table-header-cell>
            </tbody>
          </table>
        </div>
        <div class="mdc-data-table__progress-indicator">
          <div class="mdc-data-table__scrim">
          </div>
        </div>
      </div>
    </ino-table>
  </div>
</div>
`;

exports[`Structure/ino-table-header-cell Searched smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-table-header-cell">
    <ino-table class="hydrated">
      <div class="mdc-data-table ino-table--no-header">
        <div class="mdc-data-table__table-container">
          <table class="ino-table__table mdc-data-table__table">
            <tbody class="mdc-data-table__content">
              <ino-table-header-cell autofocus="false"
                                     not-sortable="false"
                                     sort-direction
                                     search-icon="search"
                                     cell-id="some-cell-id"
                                     column-id="some-random-id"
                                     searched="true"
                                     label="Label"
                                     role="columnheader"
                                     class="mdc-data-table__header-cell ino-table-header-cell--searchable ino-table-header-cell--sortable ino-table-header-cell--searched hydrated"
                                     data-column-id="some-random-id"
                                     tabindex="-1"
                                     id="some-cell-id"
                                     style="max-width: 230px;"
              >
                <ino-popover color-scheme="transparent"
                             class="hydrated"
                >
                  <div class="mdc-data-table__header-cell-wrapper"
                       slot="popover-trigger"
                       tabindex="0"
                       aria-expanded="false"
                  >
                    <ino-icon class="ino-table-header-cell__search-icon ino-icon--color-scheme-primary hydrated">
                    </ino-icon>
                    <div class="mdc-data-table__header-cell-label">
                      Label
                    </div>
                  </div>
                  <div>
                  </div>
                </ino-popover>
                <ino-icon-button class="sort-none ino-icon-button--color-scheme-dark hydrated">
                  <button class="mdc-icon-button mdc-ripple-upgraded--unbounded mdc-ripple-upgraded"
                          type="button"
                          style="--mdc-ripple-fg-size:14px; --mdc-ripple-fg-scale:1.71429; --mdc-ripple-left:5px; --mdc-ripple-top:5px;"
                  >
                    <div class="mdc-icon-button__ripple">
                    </div>
                    <span class="mdc-icon-button__icon">
                      <ino-icon class="ino-icon--color-scheme-primary hydrated">
                      </ino-icon>
                    </span>
                  </button>
                </ino-icon-button>
              </ino-table-header-cell>
            </tbody>
          </table>
        </div>
        <div class="mdc-data-table__progress-indicator">
          <div class="mdc-data-table__scrim">
          </div>
        </div>
      </div>
    </ino-table>
  </div>
</div>
`;

exports[`Structure/ino-table-header-cell SortDirection smoke-test 1`] = `
<div id="root-inner">
  <div class="story-ino-table-header-cell">
    <ino-table class="hydrated">
      <div class="mdc-data-table ino-table--no-header">
        <div class="mdc-data-table__table-container">
          <table class="ino-table__table mdc-data-table__table">
            <tbody class="mdc-data-table__content">
              <ino-table-header-cell autofocus="false"
                                     not-sortable="false"
                                     sort-direction="desc"
                                     search-icon="search"
                                     cell-id="some-cell-id"
                                     column-id="some-random-id"
                                     searched="false"
                                     label="Label"
                                     role="columnheader"
                                     class="mdc-data-table__header-cell ino-table-header-cell--searchable ino-table-header-cell--sortable hydrated"
                                     data-column-id="some-random-id"
                                     tabindex="-1"
                                     id="some-cell-id"
                                     style="max-width: 230px;"
              >
                <ino-popover color-scheme="transparent"
                             class="hydrated"
                >
                  <div class="mdc-data-table__header-cell-wrapper"
                       slot="popover-trigger"
                       tabindex="0"
                       aria-expanded="false"
                  >
                    <ino-icon class="ino-table-header-cell__search-icon ino-icon--color-scheme-primary hydrated">
                    </ino-icon>
                    <div class="mdc-data-table__header-cell-label">
                      Label
                    </div>
                  </div>
                  <div>
                  </div>
                </ino-popover>
                <ino-icon-button class="sort-desc ino-icon-button--color-scheme-dark hydrated">
                  <button class="mdc-icon-button mdc-ripple-upgraded--unbounded mdc-ripple-upgraded"
                          type="button"
                          style="--mdc-ripple-fg-size:14px; --mdc-ripple-fg-scale:1.71429; --mdc-ripple-left:5px; --mdc-ripple-top:5px;"
                  >
                    <div class="mdc-icon-button__ripple">
                    </div>
                    <span class="mdc-icon-button__icon">
                      <ino-icon class="hydrated ino-icon--color-scheme-primary">
                      </ino-icon>
                    </span>
                  </button>
                </ino-icon-button>
              </ino-table-header-cell>
            </tbody>
          </table>
        </div>
        <div class="mdc-data-table__progress-indicator">
          <div class="mdc-data-table__scrim">
          </div>
        </div>
      </div>
    </ino-table>
  </div>
</div>
`;
